src/nni_manager/core/nnimanager.ts

```diff
@@ -342,7 +341,7 @@ class NNIManager implements Manager {
 
     private async periodicallyUpdateExecDuration(): Promise<void> {
         let count: number = 1;
-        for (; ;) {
+        while (this.status.status !== 'STOPPING') {
```

Comment:
Is it possible that state transfer from 'STOPPING' to 'STOPPED' is very fast, so that this loop misses the 'STOPPING' state.
Suggest using `while (this.status.status !== 'STOPPING' && this.status.status !== 'STOPPED')` or `while (this.status.status !== 'STOPPED')`